### DISCUSSION TITLE
Error writing DSDT table: InvalidGuestAddress

### DISCUSSION THREAD

**Discussion by hw-claudio** [cloud-hypervisor/cloud-hypervisor#5015] [General]
**Describe the bug**

I am trying to build and run cloud-hypervisor, and run a simple image with CentoS7 already installed.

**To Reproduce**
Steps to reproduce the behaviour:

I followed the instructions at UEFI.md (but used the upstream tianocore/EDK2 repo), and built the CLOUDHV.fd

Then did a

rustup target add x86_64-unknown-linux-musl
cargo build --release

which succeeded.

When I start the VM through via:

./target/release/cloud-hypervisor --kernel CLOUDHV.fd --disk path=./centos7.raw --cmdline "console=hvc0 root=/dev/vda1 rw" --cpus boot=4 --memory size=4096 --net "tap=,mac=,ip=,mask="

I get:

thread 'vmm' panicked at 'Error writing DSDT table: InvalidGuestAddress(GuestAddress(655396))', vmm/src/acpi.rs:637:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error booting VM: ResponseRecv(RecvError)

**Version**

cloud-hypervisor v28.0-142-g8ab15b9a

built from source,

rustup target add x86_64-unknown-linux-musl
cargo build --release

**VM configuration**

What command line did you run (or JSON config data):

./target/release/cloud-hypervisor --kernel CLOUDHV.fd --disk path=./centos7.raw --cmdline "console=hvc0 root=/dev/vda1 rw" --cpus boot=4 --memory size=4096 --net "tap=,mac=,ip=,mask="

Guest OS version details:
CentOS 7

Host OS version details:
SUSE SLES 15 SP4

**Logs**

Output of `cloud-hypervisor -v` from either standard error or via `--log-file`:

cloud-hypervisor: 1.423344ms: <vmm> INFO:vmm/src/lib.rs:1782 -- API request event: VmCreate(Mutex { data: VmConfig { cpus: CpusConfig { boot_vcpus: 4, max_vcpus: 4, topology: None, kvm_hyperv: false, max_phys_bits: 46, affinity: None, features: CpuFeatures { amx: false } }, memory: MemoryConfig { size: 4096, mergeable: false, hotplug_method: Acpi, hotplug_size: None, hotplugged_size: None, shared: false, hugepages: false, hugepage_size: None, prefault: false, zones: None, thp: true }, payload: Some(PayloadConfig { firmware: None, kernel: Some("CLOUDHV.fd"), cmdline: Some("console=hvc0 root=/dev/vda1 rw"), initramfs: None }), disks: Some([DiskConfig { path: Some("./centos7.raw"), readonly: false, direct: false, iommu: false, num_queues: 1, queue_size: 128, vhost_user: false, vhost_socket: None, rate_limiter_config: None, id: None, disable_io_uring: false, pci_segment: 0 }]), net: Some([NetConfig { tap: None, ip: 192.168.249.1, mask: 255.255.255.0, mac: MacAddr { bytes: [46, 239, 98, 58, 125, 98] }, host_mac: None, mtu: None, iommu: false, num_queues: 2, queue_size: 256, vhost_user: false, vhost_socket: None, vhost_mode: Client, id: None, fds: None, rate_limiter_config: None, pci_segment: 0 }]), rng: RngConfig { src: "/dev/urandom", iommu: false }, balloon: None, fs: None, pmem: None, serial: ConsoleConfig { file: None, mode: Null, iommu: false }, console: ConsoleConfig { file: None, mode: Tty, iommu: false }, devices: None, user_devices: None, vdpa: None, vsock: None, iommu: false, sgx_epc: None, numa: None, watchdog: false, platform: None, tpm: None }, poisoned: false, .. }, Sender { .. })
cloud-hypervisor: 1.785566ms: <vmm> INFO:vmm/src/lib.rs:1782 -- API request event: VmBoot(Sender { .. })
cloud-hypervisor: 2.280506ms: <vmm> INFO:vmm/src/memory_manager.rs:1357 -- Anonymous mapping at 0x7f2aa3c2e000 (size = 0x1000)
cloud-hypervisor: 2.305011ms: <vmm> INFO:vmm/src/memory_manager.rs:1574 -- Creating userspace mapping: 0 -> 7f2aa3c2e000 1000, slot 0
cloud-hypervisor: 2.360825ms: <vmm> INFO:vmm/src/memory_manager.rs:1608 -- Created userspace mapping: 0 -> 7f2aa3c2e000 1000
cloud-hypervisor: 2.418272ms: <vmm> INFO:vmm/src/vm.rs:495 -- Booting VM from config: Mutex { data: VmConfig { cpus: CpusConfig { boot_vcpus: 4, max_vcpus: 4, topology: None, kvm_hyperv: false, max_phys_bits: 46, affinity: None, features: CpuFeatures { amx: false } }, memory: MemoryConfig { size: 4096, mergeable: false, hotplug_method: Acpi, hotplug_size: None, hotplugged_size: None, shared: false, hugepages: false, hugepage_size: None, prefault: false, zones: None, thp: true }, payload: Some(PayloadConfig { firmware: None, kernel: Some("CLOUDHV.fd"), cmdline: Some("console=hvc0 root=/dev/vda1 rw"), initramfs: None }), disks: Some([DiskConfig { path: Some("./centos7.raw"), readonly: false, direct: false, iommu: false, num_queues: 1, queue_size: 128, vhost_user: false, vhost_socket: None, rate_limiter_config: None, id: None, disable_io_uring: false, pci_segment: 0 }]), net: Some([NetConfig { tap: None, ip: 192.168.249.1, mask: 255.255.255.0, mac: MacAddr { bytes: [46, 239, 98, 58, 125, 98] }, host_mac: None, mtu: None, iommu: false, num_queues: 2, queue_size: 256, vhost_user: false, vhost_socket: None, vhost_mode: Client, id: None, fds: None, rate_limiter_config: None, pci_segment: 0 }]), rng: RngConfig { src: "/dev/urandom", iommu: false }, balloon: None, fs: None, pmem: None, serial: ConsoleConfig { file: None, mode: Null, iommu: false }, console: ConsoleConfig { file: None, mode: Tty, iommu: false }, devices: None, user_devices: None, vdpa: None, vsock: None, iommu: false, sgx_epc: None, numa: None, watchdog: false, platform: None, tpm: None }, poisoned: false, .. }
cloud-hypervisor: 2.534187ms: <payload_loader> INFO:vmm/src/vm.rs:951 -- Loading kernel
cloud-hypervisor: 2.754306ms: <vmm> INFO:vmm/src/cpu.rs:757 -- Request to create new vCPUs: desired = 4, max = 4, allocated = 0, present = 0
cloud-hypervisor: 2.77288ms: <vmm> INFO:vmm/src/cpu.rs:704 -- Creating vCPU: cpu_id = 0
cloud-hypervisor: 2.909745ms: <vmm> INFO:vmm/src/cpu.rs:704 -- Creating vCPU: cpu_id = 1
cloud-hypervisor: 2.957754ms: <vmm> INFO:vmm/src/cpu.rs:704 -- Creating vCPU: cpu_id = 2
cloud-hypervisor: 2.991406ms: <vmm> INFO:vmm/src/cpu.rs:704 -- Creating vCPU: cpu_id = 3
cloud-hypervisor: 3.0471ms: <vmm> INFO:vmm/src/pci_segment.rs:90 -- Adding PCI segment: id=0, PCI MMIO config address: 0xe8000000, device area [0x100000000-0x7feffffffff
cloud-hypervisor: 3.107512ms: <vmm> INFO:vmm/src/device_manager.rs:2132 -- Creating virtio-block device: DiskConfig { path: Some("./centos7.raw"), readonly: false, direct: false, iommu: false, num_queues: 1, queue_size: 128, vhost_user: false, vhost_socket: None, rate_limiter_config: None, id: Some("_disk0"), disable_io_uring: false, pci_segment: 0 }
cloud-hypervisor: 3.351916ms: <vmm> INFO:vmm/src/device_manager.rs:2210 -- Using asynchronous RAW disk file (io_uring)
cloud-hypervisor: 3.367224ms: <vmm> INFO:virtio-devices/src/block.rs:485 -- Disk topology: DiskTopology { logical_block_size: 512, physical_block_size: 512, minimum_io_size: 512, optimal_io_size: 0 }
cloud-hypervisor: 3.393223ms: <vmm> INFO:vmm/src/device_manager.rs:2307 -- Creating virtio-net device: NetConfig { tap: None, ip: 192.168.249.1, mask: 255.255.255.0, mac: MacAddr { bytes: [46, 239, 98, 58, 125, 98] }, host_mac: None, mtu: None, iommu: false, num_queues: 2, queue_size: 256, vhost_user: false, vhost_socket: None, vhost_mode: Client, id: Some("_net1"), fds: None, rate_limiter_config: None, pci_segment: 0 }
cloud-hypervisor: 4.482476ms: <vmm> INFO:vmm/src/device_manager.rs:2463 -- Creating virtio-rng device: RngConfig { src: "/dev/urandom", iommu: false }
cloud-hypervisor: 4.657762ms: <vmm> INFO:vmm/src/vm.rs:2038 -- Booting VM

Linux kernel output:

journalctl -q -a :

Dec 15 04:12:56 virt158 systemd-udevd[79591]: Using default interface naming scheme 'sle15-sp4'.
Dec 15 04:12:56 virt158 wickedd-nanny[2324]: /org/opensuse/Network/Interface/12.getManagedObjects failed. Server responds:
Dec 15 04:12:56 virt158 wickedd-nanny[2324]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.freedesktop.DBus.ObjectManager" doesn't exist
Dec 15 04:12:56 virt158 wickedd-nanny[2324]: /org/opensuse/Network/Interface/12.getManagedObjects failed. Server responds:
Dec 15 04:12:56 virt158 wickedd-nanny[2324]: org.freedesktop.DBus.Error.UnknownMethod: Method "GetManagedObjects" with signature "" on interface "org.freedesktop.DBus.ObjectManager" doesn't exist

RUST_BACKTRACE=1

Note that trying the suggestion to use RUST_BACKTRACE=1 gets me a segfault:

==== Possible seccomp violation ====
Try running with `strace -ff` to identify the cause and open an issue: https://github.com/cloud-hypervisor/cloud-hypervisor/issues/new
Bad system call (core dumped)

# gdb -core core.dump

Core was generated by `./target/release/cloud-hypervisor --kernel CLOUDHV.fd --disk path=centos7.raw -'.
Program terminated with signal SIGSYS, Bad system call.
#0  0x00007f341096871b in ?? ()
[Current thread is 1 (LWP 79708)]

(gdb) info threads
  Id   Target Id         Frame 
* 1    LWP 79708         0x00007f341096871b in ?? ()
  2    LWP 79707         warning: Section `.reg-xstate/79707' in core file too small.
0x00007f3410469ec9 in ?? ()
  3    LWP 79709         warning: Section `.reg-xstate/79709' in core file too small.
0x00007f34109679c6 in ?? ()

(gdb) disassemble 0x00007f341096871b-30,0x00007f341096871b+30
Dump of assembler code from 0x7f34109686fd to 0x7f3410968739:
   0x00007f34109686fd:  (bad)  
   0x00007f34109686fe:  push   0x3f(%rdi)
   0x00007f3410968701:  mov    %eax,%r8d
   0x00007f3410968704:  mov    $0x8,%r10d
   0x00007f341096870a:  xor    %edx,%edx
   0x00007f341096870c:  mov    %r9,%rsi
   0x00007f341096870f:  mov    $0x2,%edi
   0x00007f3410968714:  mov    $0xe,%eax
   0x00007f3410968719:  syscall 
=> 0x00007f341096871b:  mov    0x108(%rsp),%rcx

**Comment by sboeuf** [comment]
The error comes from the parameter you used in the Cloud Hypervisor command:
```
--memory size=4096
```
You basically created a VM with 4kiB of RAM :)

Try something like `--memory size=4G`, that should work better!

One more thing, when you use the `CLOUDHV.fd` firmware, the `--cmdline "console=hvc0 root=/dev/vda1 rw"` parameter is ignored.

**Reply by hw-claudio** [reply]
ooops! Thanks!
